### ISSUE TITLE
phRAIDER error

### ISSUE THREAD

**Issue by danshu** [karroje/phRAIDER#1]
Hi,

I run phRAIDER with default setting:
phRAIDER "SEQUENCE_FILE" "OUTPUT_DIRECTORY"
The error message is:
"Writing results elements...
Writing families to ./families
Writing elementary repeats to ./elements
ERROR: unable to find correct threshold
ERROR: unable to find associated sequence ID
..."

So what's the cause of this error? 

Thanks!
Quan

**Comment by karroje** [comment]
Hello Quan,

```
 Would you be willing to share your sequence file with me?  It would be
```

easier to track this down if I could run it myself.

```
 John
```

---

Dr. John Karro, Associate Professor
Department of Computer Science and Software Engineering
Affiliate: Department of Microbiology, Department of Statistics

## Office: Benton 205D, Miami University, Oxford, Ohio

**Comment by danshu** [comment]
Hello John,

Thanks for your kind reply. If there is no known solution to this error, I can try to track it down first instead of bothering you. I may turn to you later if I still can not make it.

Thanks,
Quan

**Comment by danshu** [comment]
I forgot to mention that I have already masked my sequence file with repeatmasker using repbase. Would this be the cause of this error? And I got 3 results files: families, elements and summary_info.

**Comment by danshu** [comment]
Hello John,

I have found a minimum of 5 sequences on which I can reproduce the same error while no error occurs when I run on each of the 5 sequences separately. I can share with you the sequence file (148 kb) and please tell me the way you want to receive the file.

Thanks,
Quan

**Comment by karroje** [comment]
Try running it with the following command:

   ./phRAIDER -p -q -c 2 -s 10101111100110100111011111 source_file output_dir
where <source file> is a fasta holding your genomic sequence, and <output dir> is a directory where you want your output to end up.  After the run you should find a file in that directory called elements, which is what you want.

Note: we have a new version of phRAIDER not yet released that will give you considerably better results.  If you can share your data, I would be happy to run it for you.  If not, I can share the executable with you -- though it is not yet so easy to run.

**Comment by karroje** [comment]
Yes, if you could just email me (karroje@miamioh.edu) the data that would be appreciated.

**Comment by karroje** [comment]
I've just pushed a new version that has better default parameters.  If you pull and recompile, you can now run with:

./phRAIDER -p source_file output_dir

(The -p is optional uses less memory but increases runtime.)

**Comment by simoncchu** [comment]
I met the same problem with default parameter:
"
ERROR: unable to find associated sequence ID
ERROR: unable to find correct threshold
"
The program keep on reporting this information but still running. Any suggestions?

**Comment by karroje** [comment]
Hi,

    I could use a bit more information:

* Can you give me the exactly command you are using to launch the tool?
* Does your sequence file hold one sequence or multiple sequences?
* If necessary, can you share the sequence file with me?  (Don't worry
about that yet -- but let me know if its an option.)

John

**Comment by simoncchu** [comment]
@karroje 
-The command I used: phRAIDER draft_genome.fa ./../repeats_phRAIDER/
-The "draft_genome.fa" file contains several scaffolds, so its multiple sequences
-I think the sequence file is quite large ~6G

**Comment by figuernd** [comment]
@Reedwarbler When the FASTA contains multiple sequences, RAIDER concatenates them all searches for repeats across the whole, is this what you wish for? Did this error arise immediately or after a long runtime? This error could only be fired after all elements have identified, and if it arises with no noticeable runtime on a file that large then it is likely not reading the sequence at all.

**Comment by figuernd** [comment]
The issue might actually be related to the length of the sequence: we use unsigned integers to mark element indices, which are only 4 bytes. 32 bits can only address 4GB of sequence. We used 64-bit integers in the past, but they increase memory requirements considerably.

**Comment by simoncchu** [comment]
@figuernd Does it add separators like "$$$$" when concatenate? Actually I want these each contig as separate sequence. So you suggests run it one by one?  The error appears after running for some time.

**Comment by figuernd** [comment]
@Reedwarbler Ah no but a separator would be a good idea. We just maintain an in-memory list of where each sequence ends ("thresholds") for mapping indices back to their original sequence. If you wish to identify repeats in each sequence independently of the others, you will want to run them one by one. That might resolve the issue if 4-byte addressing was in fact the issue.

**Comment by karroje** [comment]
I apologize: I missed that you had responded. I will go over this and
address in depth over the weekend.
